File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        map ={}

        for i in range(len(nums)):

            n = nums[i]
            if target-n in map:

                return [map[target-n],i]
            else:

                map[n]=i
        return [0,0]
        

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        
        HashMap<Integer,Integer> map = new HashMap<>();


        for(int i=0 ;i<nums.length;i++)
        {
            int num = nums[i];

            if(map.containsKey(num))
            {    
                return true;
            }
            else
                map.put(num,1);
        }
        return false;
    }
}